test/calibration/test_setup_library.py

```diff
@@ -211,20 +242,74 @@ def _my_build_schedules():
 
         TestLibrary._build_schedules = build_schedules
 
-    def _test_library_equivalence(self, lib1, lib2) -> bool:
-        """Test if libraries are equivalent.
 
-        Two libraries are equivalent if they have the same basis gates and
-        if the strings of the schedules are equal. We cannot directly compare
-        the schedules because the parameter objects in them will be different
-        instances.
-        """
+@ddt
+class TestFixedFrequencyTransmonCR(QiskitExperimentsTestCase):
+    """Test the cross-resonance extension of the fixed frequency library."""
 
-        if len(set(lib1.basis_gates)) != len(set(lib2.basis_gates)):
-            return False
+    def test_library(self):
+        """Basic test of the library."""
+        self.assertTrue("cr" in EchoedCrossResonance(FixedFrequencyTransmon()))
+
+    def test_exceptions(self):
+        """Test that x gate is required."""
+        with self.assertRaises(CalibrationError):
+            EchoedCrossResonance(FixedFrequencyTransmon(basis_gates=["y", "cr"]))
+
+    @data(
+        (1, 2, 2),
+        (3, 4, 6),
+    )
+    @unpack
+    def test_cals_initialization(self, control, target, uchan):
+        """Test the an instance of Calibrations can be initialized with this library."""
+        sq_lib = FixedFrequencyTransmon()
+        libraries = [sq_lib, EchoedCrossResonance(sq_lib)]
```

Comment:
Indeed, the example you show is not the intended use. You should do
```python
sq_lib = FixedFrequencyTransmon(...)
cal = Calibrations(
    ..., library=[sq_lib, EchoedCrossResonance(..., sq_lib)]
)
```
in your example you will have independent parameters and schedules.

Reply:
Perhaps worth being more explicit on the documentation.

Reply:
Or perhaps you can check duplicated gate name definition from different library instance? Then calibrations can simply raise an error. Library instance is hashable?

Reply:
Actually I think this brings us to a bug in `Calibrations`. They currently check the added schedule for call instructions and raise if the schedule key by name and number of qubits is not present. I think we might have two issues here:
* This checking is currently not done recursively.
* The checking should check the actual schedule instances and not just the keys.

Reply:
Hmm, this discussion and some other things going on in this PR make me want to revisit the `CalledSchedule` idea that we had considered previously when dealing with some of the intricacies of using `get_schedule` on a schedule with nested calls. The idea is that it would be an `Instruction` created like `CalledSchedule(name="x", channels=(d0,))` (perhaps it could instead be a `Schedule` variant with the existing `Call` instruction). When `Calibrations.get_schedule()` walks through a template schedule to do substitutions, it would replace a `CalledSchedule` with the result of `Calibrations.get_schedule()` with the appropriate name and qubits.

The main benefit I see to adding this instruction is reduced coupling between the different schedules. For instance, `EchoedCrossResonance()` can call `x` without needing to take a single qubit library as an argument, and there is no need to maintain links across libraries when serializing them. Comparing schedules is also easier since one does not need to walk recursively through subroutines (though maybe we still have to support users creating nested schedules). Also, one can modify a called schedule without needing to redefine the calling schedules -- for example, changing to a gaussian square x pulse without recreating the cross-resonance schedules.

The downside is that this instruction makes the nested  template schedule not a complete schedule. It is only complete in the context of `Calibrations.get_schedule()`, but this is already somewhat the case since the templates have unbound parameters that are bound by `get_schedule()`. Another downside is that a new instruction has to be created and perhaps that makes Calibrations work less in line with standard qiskit.